tests/test_predictions.py

```diff
@@ -20,6 +20,18 @@
     "https://raw.githubusercontent.com/vincentarelbundock/Rdatasets/master/csv/ggplot2/diamonds.csv"
 )
 
+def test_newdata_balanced():
+    df = pl.read_csv(
+        "https://vincentarelbundock.github.io/Rdatasets/csv/palmerpenguins/penguins.csv",
+        null_values="NA",
+    ).drop_nulls()
+    mod = smf.ols(
+        "body_mass_g ~ flipper_length_mm * species * bill_length_mm + island",
+        df.to_pandas(),
+    ).fit()
+    pre_py = predictions(mod, newdata="balanced")
```

Comment:
Please add a test for the number of rows. I expect that this will have 9 rows since (I believe) `species` and `island` both have 3 unique values.